src/main/java/com/hadoop/compression/lzo/DistributedLzoIndexer.java

```diff
@@ -25,67 +27,153 @@
 
 public class DistributedLzoIndexer extends Configured implements Tool {
   private static final Log LOG = LogFactory.getLog(DistributedLzoIndexer.class);
-  private final String LZO_EXTENSION = new LzopCodec().getDefaultExtension();
 
+  private static final String LZO_EXTENSION = new LzopCodec().getDefaultExtension();
+
+  public static final String LZO_INDEXING_SKIP_SMALL_FILES_KEY = "lzo.indexing.skip-small-files.enabled";
+  public static final boolean LZO_INDEXING_SKIP_SMALL_FILES_DEFAULT = false;
+  public static final String LZO_INDEXING_SMALL_FILE_SIZE_KEY = "lzo.indexing.skip-small-files.size";
+  public static final long LZO_INDEXING_SMALL_FILE_SIZE_DEFAULT = 0;
+  public static final String LZO_INDEXING_RECURSIVE_KEY = "lzo.indexing.recursive.enabled";
+  public static final boolean LZO_INDEXING_RECURSIVE_DEFAULT = true;
+  private static final String TEMP_FILE_EXTENSION = "/_temporary";
+
+  private boolean lzoSkipIndexingSmallFiles = LZO_INDEXING_SKIP_SMALL_FILES_DEFAULT;
+  private boolean lzoRecursiveIndexing = LZO_INDEXING_RECURSIVE_DEFAULT;
+  private long lzoSmallFileSize = LZO_INDEXING_SMALL_FILE_SIZE_DEFAULT;
+
+  public boolean getLzoSkipIndexingSmallFiles() { return lzoSkipIndexingSmallFiles; }
+  public boolean getLzoRecursiveIndexing() { return lzoRecursiveIndexing; }
+  public long getLzoSmallFileSize() { return lzoSmallFileSize; }
+
+  private Configuration conf = getConf();
+
+  /**
+   * Accepts paths which don't end in TEMP_FILE_EXTENSION
+   */
   private final PathFilter nonTemporaryFilter = new PathFilter() {
+    @Override
     public boolean accept(Path path) {
-      return !path.toString().endsWith("/_temporary");
+      return !path.getName().endsWith(TEMP_FILE_EXTENSION);
```

Comment:
I must have missed this in the last pass. Path.getName() returns only the final component of the path (file/directory name) whereas Path.toString() returns the full path as a string. Since TEMP_FILE_EXTENSION starts with "/", path.getName().endsWith(TEMP_FILE_EXTENSION) will never return true, so this filter is not right, correct?